crates/ironrdp-pdu/src/rdp/suppress_output.rs

```diff
@@ -0,0 +1,57 @@
+use std::io;
+
+use byteorder::{ReadBytesExt as _, WriteBytesExt as _};
+use num_derive::{FromPrimitive, ToPrimitive};
+use num_traits::{FromPrimitive, ToPrimitive};
+
+use crate::{geometry::InclusiveRectangle, PduParsing};
+
+#[repr(u8)]
+#[derive(Debug, PartialEq, Eq, FromPrimitive, ToPrimitive, Clone)]
+pub enum AllowDisplayUpdatesType {
+    SuppressDisplayUpdates = 0x00,
+    AllowDisplayUpdates = 0x01,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct SuppressOutputPdu {
+    allow_display_updates: AllowDisplayUpdatesType,
+    desktop_rect: Option<InclusiveRectangle>,
+}
+
+impl PduParsing for SuppressOutputPdu {
+    type Error = io::Error;
+
+    fn from_buffer(mut stream: impl io::Read) -> Result<Self, Self::Error> {
+        let allow_display_updates = AllowDisplayUpdatesType::from_u8(stream.read_u8()?)
+            .ok_or_else(|| io::Error::new(io::ErrorKind::InvalidData, "Invalid display update type"))?;
+        let _padding = stream.read_u8()?; // padding
+        let _padding = stream.read_u8()?; // padding
+        let _padding = stream.read_u8()?; // padding
+        let desktop_rect = if allow_display_updates == AllowDisplayUpdatesType::AllowDisplayUpdates {
+            Some(InclusiveRectangle::from_buffer(&mut stream)?)
+        } else {
+            None
+        };
+        Ok(Self {
+            allow_display_updates,
+            desktop_rect,
+        })
+    }
+
+    fn to_buffer(&self, mut stream: impl io::Write) -> Result<(), Self::Error> {
+        stream.write_u8(self.allow_display_updates.to_u8().unwrap())?;
+        stream.write_u8(0)?; // padding
+        stream.write_u8(0)?; // padding
+        stream.write_u8(0)?; // padding
+        if let Some(rect) = &self.desktop_rect {
+            rect.to_buffer(&mut stream)?;
+        }
+
+        Ok(())
+    }
+
+    fn buffer_length(&self) -> usize {
+        4 + self.desktop_rect.as_ref().map_or(0, |r| r.buffer_length())
```

Comment:
Note (style): FYI, we are following [this convention](https://github.com/Devolutions/IronRDP/blob/d69ff6e6e7ebf76864f4d812a33677b8d0d70eac/STYLE.md#formatting-for-sizes--lengths-eg-in-encodesize-and-fixed_part_size-definitions)

In this case, it could be something like this:
```rust
1 // allowDisplayUpdates
+ 3 // pad3Octets
+ self.desktop_rect.as_ref().map_or(0, |r| r.buffer_length()) // desktopRect
```

Reply:
ok